src/plcc/load_spec/load_rough_spec/parse_dividers.py

```diff
@@ -7,14 +7,69 @@
 
 @dataclass
 class Divider:
+    tool: str
+    language: str
     line: Line
 
+def parse_dividers(lines: list[Line], tool='Java', language='Java') -> list[Divider | Line]:
+    return ParseDividersHelper(lines, tool, language).parse()
 
-def parse_dividers(lines, pattern=re.compile(r'^%(?:\s.*)?$'), Divider=Divider):
-    if lines is None:
-        return
-    for line in lines:
-        if isinstance(line, Line) and pattern.match(line.string):
-            yield Divider(line)
+class ParseDividersHelper:
```

Comment:
Rename to DividerParser.

"Helper" is vague. How does it help? In what way? Once you answer that, put it in the name.